Fixes validation of ampq credentials for OpenStack

petrblaho opened the pull request:
Original problem was when verify_credentials method returned false
code in controller went into else branch of begin rescue block and
rendered "successful" flash message.

This patch adds if-else to test result of verify_credentials method call
and renders "unsuccessful" flash message if the result is false.

Also adds tests for this behavior.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1136289

petrblaho commented:
I tested it agains openstack providers.

I run through code to see if I do not break any other providers but everything looks good.

Please, test against other providers.

Fryguy commented:
Doesn't (or maybe, shouldn't) the verification be done in a queue item and not synchronously in the UI? cc @h-kataria

h-kataria commented:
@fryguy currently verification of credentials is being done synchronously in UI.

petrblaho commented:
Corrected what @miq-bot complained about.

@Fryguy this is not question for me, probably :-) I not sure about UX design in this app, I just tried to correct a bug. But I think that user should get response wether credentials are valid or not immediately after she submitted form. I am not sure about other options of doing this available in manageiq other than this. Generally I changed just this part of code and did not dive into researching how to do the whole feature other way.

martinpovolny commented:
@blomquisg, @petrblaho +1 on that. Maybe we can create an issue for this being done async. But it's out of scope for this fix.
